app/data/smiles.py

```diff
@@ -40,22 +41,34 @@ def construct_smiles(csv):
   return smiles, include_mpo
 
 def filter_smiles(good_smiles, smiles):
-  return {smi: data for (smi, data) in smiles.items() if smi in good_smiles}
+  return {smi: data for (smi, data) in smiles.items() if data['smart'] in convert_array_of_smarts_to_smiles(good_smiles)}
 
 def convert(mol_smile):
     m = Chem.MolFromSmiles(mol_smile)
     core = MurckoScaffold.GetScaffoldForMol(m)
     return Chem.MolToSmiles(core)
 
+def convertToSmart(chemical):
+  mol = Chem.MolFromSmiles(chemical)
+  return Chem.MolToSmiles(mol)
+
 def convert_from_smart(smart):
   conv = Chem.MolFromSmiles(smart)
   back = Chem.MolToSmiles(conv)
   return back
 
+def convert_array_of_smarts_to_smiles(smarts):
+  result = []
+
+  for smart in smarts:
+    smile = convert_from_smart(smart)
+    result.append(smile)
+
+  return result
```

Comment:
nit: this could be simplified to:
`return [convert_from_smart(smart) for smart in smarts]`